core/src/test/scala/unit/kafka/security/auth/SimpleAclAuthorizerTest.scala

```diff
@@ -92,6 +92,13 @@ class SimpleAclAuthorizerTest extends ZooKeeperTestHarness {
     simpleAclAuthorizer.authorize(session, Read, Resource(Topic, "something", PREFIXED))
   }
 
+  @Test
+  def testAuthorizeWithEmptyResourceName(): Unit = {
+    assertFalse(simpleAclAuthorizer.authorize(session, Read, Resource(Group, "", LITERAL)))
+    simpleAclAuthorizer.addAcls(Set[Acl](allowReadAcl), Resource(Group, WildCardResource, LITERAL))
+    assertTrue(simpleAclAuthorizer.authorize(session, Read, Resource(Group, "", LITERAL)))
```

Comment:
Is it possible to authorize the empty resource only? It would be good to have a test for that case too.

Reply:
I don't think we can create ACLs for empty string since ACLs are stored in a ZK node with the resource name (it is an odd default to use, but I suppose we can't get of it now).

Reply:
Right, I actually tried testing this but the create failed with an exception because of the reason you mention.

Reply:
Maybe we should just have a test for that case to assert the current behaviour (even if not ideal)